### ISSUE TITLE
Fate/ Grand Order NA detects root

### ISSUE THREAD

**Issue by istir** [topjohnwu/Magisk#502]
Few days ago FGO NA rolled another update which prevents users with some custom ROMs (like AOSP Extended) and also all rooted users from playing the game. Magisk manager is repackaged, Magisk Hide is also set up. On Pixel Experience (Android P) game works fine without magisk, but when magisk is flashed it works for few seconds and then crashes as if root was present.

**Comment by lijun0326** [comment]
#513 same issue

**Comment by ArbitraryFox** [comment]
This happens because Adaway hosts are now detected on game launch even if you enable the systemless method in Magisk. After removing it, the game will boot normally but it still force closes in the loading segment. The only way to fix it is to remove magisk.

**Comment by lijun0326** [comment]
@ArbitraryFox Why the hell they are checking hosts...

**Comment by ArbitraryFox** [comment]
@lijun0326 
Probably because it's a known indicator of having root access at one point.

**Comment by istir** [comment]
I'm currently unrooted on android p but I have adaway hosts installed and game works fine so your explanation is wrong.

**Comment by ArbitraryFox** [comment]
@istir I'm on an old N ROM and I have tested this myself.
But to give the benefit of the doubt, I will test it again with Magisk removed and will update this post. Thanks for the info anyhow.

**Comment by istir** [comment]
@ArbitraryFox 
I'm not sure how it is on N but when I was using AOSP Extended Oreo on OnePlus 5 I couldn't run the game at all without Magisk and I haven't tried without modified hosts. Next day I flashed PixelExperience Pie and game works great after flashing fgo_patch.zip. I also installed Magisk and game opens but crashes around first "servant" loading screen (10 second in). After unrooting everything works flawlessly. I still have to root every few days because I have permissive kernel and Fire Emblem Heroes doesn't like that, but after I do what I want in FEH, unroot I can play FGO normally. 

Try flashing [fgo_patch.zip](https://drive.google.com/file/d/0B8e7Gjb1XkJ0eHozYmFoTHJFcEU/view?usp=sharing), it's from [this thread on reddit](https://www.reddit.com/comments/6kcuih). It changes kernel flag ro.debuggable to 0 and build.prop to ro.build.type=user. 

It should work on any device but if something goes wrong it's not my fault 😃. But it can be very easily reverted by simply flashing ROM again or changing build.prop and flashing some kernel.

**Comment by ArbitraryFox** [comment]
@istir 
I know that patch, I automated all the changes in my kernel (that's how I had all root features on an unrooted ROM) but still no luck.
Also, I can confirm that modified host file does not trigger detection. That means FGO is detecting Magisk functions from there.

**Comment by 11lb8oz** [comment]
Lol

**Comment by istir** [comment]
@roseyhead oh hell yeah, it works! I would never even consider thinking about changing date. Thanks bud, let's hope for a magisk update next!

**Comment by istir** [comment]
Well looks like 17.1 didn't fix it either. Good thing @roseyhead fix works

**Comment by Ricardo1991** [comment]
17.1 does not work because nothing was done to make it work. The dev said he would look into it. FGO added some new library that can detect it, its unrelated to what Magisk did before with safety net

**Comment by hparadiz** [comment]
Still broken on 19.3

Figured I'd bump this for visibility.

The time thing didn't work either. Running a custom Pie ROM on a Galaxy S10+

**Comment by squaresmile** [comment]
> Still broken on 19.3

It's not broken on 19.3. Definitely works on mine and many others' phones.

> a custom Pie ROM

Depending on the custom ROM, FGO might detects it. Make sure you look up the right error code.

**Comment by reconman** [comment]
@hparadiz Works fine on my OnePlus 6 on Magisk 19.3.

**Comment by hparadiz** [comment]
It's error code 61

**Comment by reconman** [comment]
@hparadiz Have you:

1. Changed the Manager's app name in the settings?
2. Activated Magisk Hide for FGO?
3. Disabled Developer Mode in Android?

**Comment by hparadiz** [comment]
Because of the really weird way that root works on the Galaxy 10 devices Magisk is actually patched into the recovery partition so I'm not sure if I can mess with Manager.

Link here for the curious: https://forum.xda-developers.com/galaxy-s10/development/recovery-twrp-3-3-0-galaxy-s10-exynos-t3924856

Not really sure what the right approach here is. I'll play with it though. Thanks for confirming that it works though.

**Comment by hparadiz** [comment]
Okay yea trying to hide the manager's app name in settings reinstalls Manager with a random AppID but I can't remove the existing one because it's flashed to recovery and then linked to the system partition. `¯\_(ツ)_/¯`

**Comment by ArbitraryFox** [comment]
@hparadiz 
Check your build.prop, make sure ro.build.type= user and not usersebug.

Did you enable magisk manager repackage?
Magisk hide google services framework and play store too.

Have you tried this?
https://www.google.com/amp/s/amp.reddit.com/r/grandorder/comments/6kcuih/guide_to_play_fgo_on_unrooted_lineageos/

I've been running fgo on a pie rom, nothing funky going on.

**Comment by hparadiz** [comment]
I got it working.

The way the Galaxy S10+ is rooted requires flashing Magisk to recovery so when Magisk hides itself it successfully installs a copy of itself in system but fails to remove the old one. I manually removed the old one with root terminal and also had to remove the /data/data directory. Fate Go and Pokemon Go both work now. Issue was figuring out all the locations to remove magisk manager.

Thanks everyone.

![image](https://user-images.githubusercontent.com/195216/62478876-31beed00-b77a-11e9-876a-1b22d244f87d.png)
